README.md

```diff
@@ -1,205 +1,278 @@
+
+Each and every function should have a test. Our goal is 100% code coverage. As you write tests, check that they are working by with Ctrl/Cmd-Shift-T or devtools::tests().
+
+### C Code Standards <a name="cinfo"/>
+
+#### Style Guide <a name="cstyle"/>
+
+#### Code Development <a name="cdevel"/>
+
+#### Documentation  <a name="cdoc"/>
+
+We use Doxygen to write documentation in C. The style guide and rules for Doxygen can be found [here](http://www.stack.nl/~dimitri/doxygen/).
+
+To create a document from your Doxygen code, navigate to the desired directory (i.e. 'Git/SOILWAT2'), and then simply execute the 'Doxygen' command in the terminal along with the configuration file: `Doxygen Doxyfile`
+This will create a index.html file in /doc folder that you can open in any web browser.
+
+Refer to our [C documentation checklist](CDocChecklist.md) each time you are plan to commit changes with alterations to C documentation.
+
+### Unit Testing <a name="ctest"/>
+
+We use [googletest](https://github.com/google/googletest/blob/master/googletest/docs/primer.md) for unit testing in C. Within our C respoistories there is a /test folder and a /googletest folder. The /googletest folder is a submodule link to the googletest functionality and should not be editted. In the /test folder, for each .c file there is a corresponding test_.cc file which contains the unit tests for the functions found in the .c file. For example, for the SOILWAT2/SW_Site.c file there is a SOILWAT2/test/test_SW_Site.cc file for unit tests.
+
+Each and every function should have a test. Our goal is 100% code coverage. As you write tests, check that they are working by running `make test_clean test test_run` in the terminal.
+
+If your unit tests fail, begin to question whether your unit test is wrong, or whether the code is wrong. Both are possible. In C, through adding unit tests we have found many instances of failing unit tests due to memory leaks and misappropriated indices.
+
+#### Formatting of unit tests
+
+Please refer to our style guide for [unit test formatting](CUnitTestFormat.md). The key here is consistency and neatness.
+
+Make sure your unit tests are formatted correctly before you request code review.
+
+#### Conditions and assertions
+
+* For functions that calculate related vales across layers (nlyrs) or simulation (nRgr) you should always test when these each each one (i.e. nlyr = 1) and under the maximum conditions (n_lyr = MAX_N_LYR) if possible. All unit tests should be run twice under each of these conditions.
+* An if or if else statement should typically be tested, if possible.
+* Never calculate an expected answer within the TEST and test an output against it.
```

Comment:
Don't use a random number generator to create input values -- unless specifically required.